app/routes/routes.py

```diff
@@ -20,23 +49,90 @@ def FAQ_site():
     return render_template("main/FAQ.html")
 
 
-"""
-@app.route("/user_homepage", methods=["GET", "POST"])
+@app.route("/user_homepage", methods=["GET"])
+@login_required
 def user_homepage():
-    logout_user()
-    return redirect(url_for("home"))
-"""
+    robots_list = get_robots_for_user(current_user)
+    if not robots_list:
+        generate_random_robots(current_user, number_of_robots=5000)
+        robots_list = get_robots_for_user(current_user)
+    matched_robots = get_matched_robots(
+        current_user.id, current_user.name, include_chatroom_id="yes"

```

Comment:
```suggestion
        current_user.id, current_user.name, include_chatroom_id=True
```

why just don't use a boolean here?